### PR TITLE
breaking change: stats spec, spec repo, stream to value on files read

### PR DESCRIPTION
**Pull request by hacdias**
Ref.: https://github.com/ipfs/interface-ipfs-core/pull/208. 

- tests: STAT spec
- fix(breaking change): REPO spec
- feat(breaking change): FILES spec: stream to value on files read

### CODE REVIEW

**Comment by diasdavid** [comment]
interface-ipfs-core changes merged and released. Have fun! :)

**Comment by diasdavid** [comment]
Also, make sure to update js-ipfs ;)

**Comment by hacdias** [comment]
@diasdavid I'll update the interface-ipfs-core version here to try it out.

**Comment by hacdias** [comment]
@diasdavid the tests for STATS are passing ;)

File: package.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     "eslint-plugin-react": "^7.5.1",
     "go-ipfs-dep": "^0.4.13",
     "gulp": "^3.9.1",
-    "interface-ipfs-core": "~0.41.0",
```
Comment (diasdavid) [inline package.json]: Use `~`

Comment (hacdias) [reply package.json]: Done!

**Comment by hacdias** [comment]
@diasdavid anything missing here?

**Comment by diasdavid** [comment]
CI is not happy. Mind checking what's up? Seems that a rebase is needed.

File: README.md [hunk 2]
```diff
@@ -172,24 +172,25 @@ $ ipfs config --json API.HTTPHeaders.Access-Control-Allow-Methods "[\"PUT\", \"P
...
   - [`ipfs.block.put(block, cid, [callback])`](https://github.com/ipfs/interface-ipfs-core/tree/master/SPEC/BLOCK.md#put)
   - [`ipfs.block.stat(cid, [callback])`](https://github.com/ipfs/interface-ipfs-core/tree/master/SPEC/BLOCK.md#stat)
 
-- repo
-  - `ipfs.repo.stat()`
-  - `ipfs.repo.gc()`
+- [repo](https://github.com/ipfs/interface-ipfs-core/tree/master/SPEC/REPO.md)
+  - [`ipfs.repo.gc([options, callback])`](https://github.com/ipfs/interface-ipfs-core/tree/master/SPEC/REPO.md#gc)
+  - [`ipfs.repo.stat([options, callback])`](https://github.com/ipfs/interface-ipfs-core/tree/master/SPEC/REPO.md#stat)
+  - [`ipfs.repo.version([callback])`](https://github.com/ipfs/interface-ipfs-core/tree/master/SPEC/REPO.md#version)
```
Comment (diasdavid) [inline README.md]: This should be under the `Node Management` to match the structure in interface-ipfs-core

**Comment by hacdias** [comment]
- **Jenkins** seems to be failing on some random tests not related to this PR.
- **Circle** passed on all NodeJS tests and got a 'JavaScript heap out of memory' for the browser.
- **Appveyor**... well... Windows 32bits.
- **Travis** [some are failing with ' JavaScript heap out of memory' on browser. Nodejs tests are passing.](https://travis-ci.org/ipfs/js-ipfs-api/jobs/333629120)

**Comment by hacdias** [comment]
@diasdavid I'd like to see everything passing, but I'm getting a `"JavaScript heap out of memory"` error. What should I do?

**Comment by diasdavid** [comment]
The "out of memory" issue started with the ipfsd-ctl update. Essentially what is happening is that WebPack is trying to bundle too much code (more than one version of ipfs-api) and that is crashing the whole process. There is a way to increase the amount of memory allocated. But the good way to do it is to avoid creating the issue.

I'm going to merge this PR but will create a issue to track this.

**Comment by hacdias** [comment]
Thanks!

**Pull request merged**